app-frontend/src/app/components/histogram/quickEditHistogram/index.js

```diff
@@ -0,0 +1,289 @@
+                      dataRange: this.histogramOptions.range,
+                      baseScheme: this.histogramOptions.baseScheme
+                  }
+                : {}
+        );
+        if (this.renderDefinition) {
+            this.breakpoints = breakpointsFromRenderDefinition(
+                this.renderDefinition,
+                this.uuid4.generate
+            );
+        }
+        const ranges = get(changes, 'ranges.currentValue');
+        if (plot || metadata || ranges) {
+            const update = {
+                plot: plot || this.plot,
+                renderDefinition: renderDefinition || this.renderDefinition,
+                ranges: ranges || this.ranges
+            };
+            if (update.plot && update.renderDefinition && update.ranges) {
+                this.updateGraph(update.plot, update.renderDefinition, update.ranges);
+            }
+        }
+    }
+
+    getGraph() {
+        if (this.graphId) {
+            return this.graphService.getGraph(this.graphId);
+        }
+        return this.$q.reject();
+    }
+
+    initGraph() {
+        this.options = {
+            masks: {
+                min: false,
+                max: false
+            },
+            scale: 'SEQUENTIAL',
+            height: 50,
+            style: {
+                height: '50px'
+            },
+            type: 'single'
+        };
+        const elem = this.$element.find('.graph-container svg')[0];
+        this.graphService.register(elem, this.graphId, this.options);
+    }
+
+    onMaskChange(mask, value) {
+        if (this.options) {
+            if (!this.options.masks) {
+                this.options.masks = { min: false, max: false };
+            }
+            this.options.masks[mask] = value;
+        }
+        if (this.plot && this.renderDefinition && this.ranges) {
+            this.renderDefinition = renderDefinitionFromState(this.options, this.breakpoints);
+            this.updateGraph(this.plot, this.renderDefinition, this.ranges);
+            const metadata = Object.assign({}, this.metadata, {
+                renderDefinition: this.renderDefinition,
+                manualRenderDefinition: true
+            });
+            this.onChange({ metadata });
+        }
+    }
+
+    updateGraph(plot = this.plot, renderDefinition = this.renderDefinition, ranges = this.ranges) {
+        const currentUpdate = this.getGraph().then(graph => {
+            if (currentUpdate === this.currentUpdate) {
+                graph
+                    .setData({
+                        histogram: plot,
+                        breakpoints: this.breakpoints
+                    })
+                    .setOptions(
+                        Object.assign({}, this.options, {
+                            range: ranges.bufferedRange
+                        })
+                    )
+                    .render();
+            }
+        });
+        this.currentUpdate = currentUpdate;
+        return currentUpdate;
```

Comment:
This function should be able to be called from inside `onChanges`, which is called when changes to bindings are made but before they are bound to the controller.
IE in `onChanges` `this.var !== changes.var.currentValue`, and you don't want to assign `this.var` to `changes.var.currentValue` since that's redundant and I think will introduce a data-cycle.